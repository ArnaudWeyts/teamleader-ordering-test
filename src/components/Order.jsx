import React, { Component } from 'react';
import PropTypes from 'prop-types';

import orders from '../data/orders.json';
import ProductList from './ProductList';

class Order extends Component {
  constructor(props) {
    super(props);

    this.state = {
      order: null
    };

    this.adjustTotal = this.adjustTotal.bind(this);
  }

  componentDidMount() {
    const {
      match: {
        params: { id }
      }
    } = this.props;

    // Get order from list, replace with possible API request
    const order = orders.find(x => x.id === id);

    this.setState({ order });
  }

  adjustTotal(unitPrice) {
    const { order } = this.state;

    order.total -= unitPrice;
    order.total = order.total.toFixed(2);

    this.setState({ order });
  }

  render() {
    const { order } = this.state;

    if (!order) {
      return <div>Loading...</div>;
    }

    return (
      <div>
        <h2>Order #{order.id}</h2>
        <h2>Items</h2>
        <ProductList items={order.items} adjustTotal={this.adjustTotal} />
        <button type="button" onClick={() => console.log('add item')}>
          Add an item
        </button>
        <h2>Total: {order.total}</h2>
        <button type="button" onClick={() => console.log('order placed')}>
          Place order
        </button>
      </div>
    );
  }
}

Order.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Order;
