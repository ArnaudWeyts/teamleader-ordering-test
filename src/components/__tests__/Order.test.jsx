import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Order from '../Order';

describe('Order tests', () => {
  const order = { id: '1' };
  it('renders without crashing', () => {
    shallow(<Order order={order} />);
  });

  it('renders order', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/', '/orders']}>
        <Order order={order} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
