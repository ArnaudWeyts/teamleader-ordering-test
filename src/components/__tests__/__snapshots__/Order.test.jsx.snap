// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order tests renders order 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <h3>
      Order #
      1
    </h3>
    <a
      href="/orders/1"
      onClick={[Function]}
    >
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <span>
          View
        </span>
      </button>
    </a>
  </div>
</div>
`;
